### PR TITLE
openrgb: 0.8 -> 0.9

### PR DESCRIPTION
**Pull request by panda2134**
## Description of changes

https://gitlab.com/CalcProgrammer1/OpenRGB/-/tags/release_0.9

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Tungsten842** [comment]
I think that you should also update: [openrgb-plugin-effects](https://search.nixos.org/packages?channel=unstable&from=0&size=50&sort=relevance&type=packages&query=openrgb) and [openrgb-plugin-hardwaresync](https://search.nixos.org/packages?channel=unstable&show=openrgb-plugin-hardwaresync&from=0&size=50&sort=relevance&type=packages&query=openrgb)

**Comment by panda2134** [comment]
> I think that you should also update: [openrgb-plugin-effects](https://gitlab.com/OpenRGBDevelopers/OpenRGBEffectsPlugin/-/releases) and [openrgb-plugin-hardwaresync](https://gitlab.com/OpenRGBDevelopers/OpenRGBHardwareSyncPlugin/-/releases)

Updated with [3727072](https://github.com/NixOS/nixpkgs/pull/250358/commits/3727072bdfb2d416a0637c2f5b96a6fdb04209a5) and [54efceb](https://github.com/NixOS/nixpkgs/pull/250358/commits/54efcebf9b65f6b6f53910ef55e897f48083e9d7)

**Comment by khaneliman** [comment]
Do I need to do anything else in particular to test this package? I get ```Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)``` at runtime.

**Comment by panda2134** [comment]
> Do I need to do anything else in particular to test this package? I get `Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)` at runtime.

Did you see this when running `openrgb`? I failed to reproduce that.

**Comment by khaneliman** [comment]
> > Do I need to do anything else in particular to test this package? I get `Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)` at runtime.
> 
> Did you see this when running `openrgb`? I failed to reproduce that.

Yeah, just did a nixpkgs-review pr 250358 and ./results/openrgb/bin/openrgb

**Comment by khaneliman** [comment]
Result of `nixpkgs-review pr 250358` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>openrgb</li>
    <li>openrgb-plugin-effects</li>
    <li>openrgb-plugin-hardwaresync</li>
    <li>openrgb-with-all-plugins</li>
  </ul>
</details>

**Comment by panda2134** [comment]
> EDIT: well i get the same result when building from master so just ignore my runtime result executing the binary directly, i guess.

Yeah that could be the reason. I'm testing with `nixpkgs-unstable` which is a bit newer than `nixos-unstable`, so the Qt version might differ.

File: pkgs/applications/misc/openrgb-plugins/hardwaresync/default.nix [hunk 1]
```diff
@@ -14,28 +14,15 @@
...
   src = fetchFromGitLab {
     owner = "OpenRGBDevelopers";
     repo = "OpenRGBHardwareSyncPlugin";
     rev = "release_${version}";
-    hash = "sha256-P+IitP8pQLUkBdMfcNw4fOggqyFfg6lNlnSfUGjddzo=";
+    hash = "sha256-3sQFiqmXhuavce/6v3XBpp6PAduY7t440nXfbfCX9a0=";
   };
 
-  patches = [
-    (fetchpatch {
```
Comment (fgaz) [inline pkgs/applications/misc/openrgb-plugins/hardwaresync/default.nix:27]: `fetchpatch` is unused now, remove it from the inputs

**Review by fgaz — CHANGES_REQUESTED** [review]
Please squash 9dc89cabff69f37572748392dc39afc64a93c5ef into 54efcebf9b65f6b6f53910ef55e897f48083e9d7

**Comment by fgaz** [comment]
> Do I need to do anything else in particular to test this package?
> 
> ```
> Cannot mix incompatible Qt library (5.15.9) with this library (5.15.10)
> Aborted (core dumped)
> ```

Works for me on NixOS 23.05

**Review by fgaz — APPROVED** [review]

**Comment by SuperSamus** [comment]
#242899 needs to be closed.

**Pull request merged**